federation-js/src/composition/DirectiveMetadata.ts

```diff
@@ -192,3 +179,23 @@ export class DirectiveMetadata {
     }
   }
 }
+
+function mergeDirectiveUsages(
```

Comment:
I suspect that this does fix the bug. But I also find the semantics of this function confusing.  It looks like if `destination` is defined, then it *both* mutates directiveUsages in place and *also* returns the merged version. I feel like this is harder to think about than it it treats its arguments as immutable and just returns a merged version. Reading the call site (which makes it clear that it has a return value) I would not guess that it mutates its first argument.  (Or is it actually important that it mutates its argument?)

Reply:
e6c6c12